### PR TITLE
refactor: more lints and docs

### PR DESCRIPTION
**Pull request by doinkythederp**
## Describe the changes this PR makes. Why should it be merged?

Adds more lints and then resolves them:
- [x] Adds must_use to constructors and getters
- [x] Documents losses in precision and uses `u32::from` and equivalents to prevent accidental changes that will cause losses in precision
- [x] Adds semicolons to places that were missing semicolons where it makes sense to add them
- [x] Documents the errors returned by functions that return `Result`s
   - [x] ADI devices
   - [x] Smart devices
   - [x] Controller
   - [x] Screen
- [x] Adds separators to RGB hexes to improve readability
- [x] `BannerTheme` is now `Copy`
- [x] A few other minor changes suggested by Clippy
- [x] Fixes a bug in which `port.validate_expander()` was not called
- [x] Changes instances of "Gets x" terminology to "Returns x" in getter docs.
- [x] (breaking) `OpticalSensor::led_brightness` is now returned as a number from `[0.0, 1.0]` to match `OpticalSensor::set_led_brightness`.

### CODE REVIEW

**Comment by doinkythederp** [comment]
TODO: update changelog

File: packages/vexide-core/src/float/mod.rs [hunk 1]
```diff
@@ -65,46 +65,54 @@ pub trait Float: Sized {
     /// Returns the largest integer less than or equal to `self`.
     ///
     /// This function always returns the precise result.
+    #[must_use]
```
Comment (Tropix126) [inline packages/vexide-core/src/float/mod.rs:68]: Think we should probably have a message here. This is `libstd`'s:

`#[must_use = "method returns a new number and does not mutate the original value"]`

**Review by Tropix126 — CHANGES_REQUESTED** [review]
This in general seems good, although there are some places we might want an additional message with `#[must_use]`

**Review by Tropix126 — APPROVED** [review]
LGTM

**Pull request merged**